pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -0,0 +1,141 @@
+	CRDName = "cnsvolumeoperationrequests.cns.vmware.com"
+	// CRDSingular represent the singular name of cnsvolumeoperationrequest CRD
+	CRDSingular = "cnsvolumeoperationrequest"
+	// CRDPlural represent the plural name of cnsvolumeoperationrequest CRD
+	CRDPlural = "cnsvolumeoperationrequests"
+)
+
+// InitVolumeOperationRequestInterface creates the CnsVolumeOperationRequest
+// definition on the API server and returns an implementation of
+// VolumeOperationRequest interface. Clients are unaware of the implementation
+// details to read and persist volume operation details.
+// This function is not thread safe. Multiple serial calls to this function will
+// return multiple new instances of the VolumeOperationRequest interface.
+// TODO: Make this thread-safe and a singleton.
+func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRequest, error) {
+	log := logger.GetLogger(ctx)
+
+	// Create CnsVolumeOperationRequest definition on API server
+	log.Info("Creating cnsvolumeoperationrequest definition on API server")
+	err := k8s.CreateCustomResourceDefinitionFromSpec(ctx, CRDName, CRDSingular, CRDPlural,
+		reflect.TypeOf(v1alpha1.CnsVolumeOperationRequest{}).Name(), v1alpha1.SchemeGroupVersion.Group, v1alpha1.SchemeGroupVersion.Version, apiextensionsv1beta1.NamespaceScoped)
+	if err != nil {
+		log.Errorf("failed to create cnsvolumeoperationrequest CRD with error: %v", err)
+	}
+
+	// Get in cluster config for client to API server
+	config, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to get kubeconfig with error: %v", err)
+		return nil, err
+	}
+
+	// Create client to API server
+	k8sclient, err := k8s.NewClientForGroup(ctx, config, v1alpha1.SchemeGroupVersion.Group)
+	if err != nil {
+		log.Errorf("failed to create k8sClient with error: %v", err)
+		return nil, err
+	}
+
+	// Initialize the operationRequestStoreOnETCD implementation of VolumeOperationRequest
+	// interface.
+	// NOTE: Currently there is only a single implementation of this interface.
+	// Future implementations will need modify this step.
+	operationRequestStore := &operationRequestStoreOnETCD{
+		k8sclient: k8sclient,
+	}
+
+	return operationRequestStore, nil
+}
+
+// GetRequestDetails returns the details of the operation on the volume
+// that is persisted by the VolumeOperationRequest interface, by querying
+// API server for a CnsVolumeOperationRequest instance with the given
+// name.
+// Returns an error if any error is encountered while attempting to
+// read the previously persisted information from the API server.
+// If no previous information has been persisted, it returns an empty object.
+func (or *operationRequestStoreOnETCD) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
+	log := logger.GetLogger(ctx)
+	// TODO: Implement GetRequestDetails for operationRequestStoreOnETCD
+	err := fmt.Errorf("GetRequestDetails not implemented for operationRequestStoreOnETCD")
```

Comment:
start all error messages with lower case.

Reply:
Actually according to https://github.com/golang/go/wiki/CodeReviewComments#error-strings proper nouns should be capitalized.